src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -36,14 +55,62 @@ public void setup() {
     testUser.setPassword("1234");
     testUser.setGender(Gender.OTHER);
 
-    // when -> any object is being save in the userRepository -> return the dummy
+    //other user for matching etc
+    otherUser = new User();
+    otherUser.setId(1L);
```

Comment:
Don't use the same user id twice!

Reply:
very true

Reply:
> > Generally very nice. Why don't you use Mockito.any() anymore? Does this have a reason? I will implement a method in the Pair to see if two pairs are equal, but you don't have to change anything for that. Most of my comments are not critical, except for the one on line 60 (I think)
> 
> I try to avoid Mockito.any() and replace with more specific calls whenever possible just to be sure that it only returns for example testUser if the testUser.id is passed and not if any other number is passed..

Makes sense